### ISSUE TITLE
--env argument in quotes isn't parsed - Error: Unknown option

### ISSUE THREAD

**Issue by bes** [webpack/webpack-cli#3398]
**Describe the bug**

I have a script that generates a command line for `webpack-cli` and some of the `--env` options are quoted, e.g: `"--env favicons"`. `webpack-cli` seems to be able to handle quoting of other arguments, e.g: `webpack-cli "--help"` works just fine.

**What is the current behavior?**

My current version of `webpack-cli` is `4.10.0`, which doesn't work, but my previous version `3.3.12` did work.

**To Reproduce**

Give any `--env` argument to `webpack-cli` in quotes: `webpack-cli "--env something"`. Full example:

```zsh
NODE_ENV=production TS_NODE_PROJECT="tsconfig-for-webpack-config.json" node_modules/.bin/webpack-cli --config app/webpack.config.ts --env dev --bail '--env favicons' '--env version="v1.1.1"' '--env sentry'
```

**Expected behavior**

Quotes (single, double) should be "ignored", as is usual in bash/zsh/etc.

**Screenshots**

**Please paste the results of `npx webpack-cli info` here, and mention other relevant information**

```
  System:
    OS: macOS 12.5.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 5.14 GB / 32.00 GB
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 103.0.2
    Firefox Nightly: 87.0a1
    Safari: 15.6.1
  Packages:
    copy-webpack-plugin: ^11.0.0 => 11.0.0
    css-loader: ^6.7.1 => 6.7.1
    favicons-webpack-plugin: ^5.0.2 => 5.0.2
    fork-ts-checker-webpack-plugin: ^7.2.13 => 7.2.13
    html-webpack-plugin: ^5.5.0 => 5.5.0
    less-loader: ^11.0.0 => 11.0.0
    source-map-loader: ^4.0.0 => 4.0.0
    style-loader: ^3.3.1 => 3.3.1
    swc-loader: ^0.2.3 => 0.2.3
    warnings-to-errors-webpack-plugin: ^2.3.0 => 2.3.0
    webpack: ^5.74.0 => 5.74.0
    webpack-bundle-analyzer: ^4.6.1 => 4.6.1
    webpack-cli: ^4.10.0 => 4.10.0
    webpack-dev-server: ^4.10.1 => 4.10.1
    webpack-node-externals: ^3.0.0 => 3.0.0
    webpack-shell-plugin-next: ^2.2.2 => 2.2.2
```

**Comment by alexander-akait** [comment]
Sorry, we can't fix/improve it in webpack-cli, if you want to implement this feature, please open an issue in `commander` repo https://github.com/tj/commander.js/, we use this package to parse arguments.

But to be honest, I'm against it. It's pretty ambiguous. When you write `--bail '--env favicons'`, what does it mean?
1. `--bail` flag with `'--env favicons'` value (string value)
2. `--bail` flag and `--env` flag with `favicons` value


Also some flags can supports different values - no value, string value, raw value (i.e. `--flag`, `--flag "string"` and `--flag value`) and it can have different semantic. So I recommend avoid it. Feel free to feedback.

**Comment by bes** [comment]
Thank you for your response.

Normally unescaped quotes are just handled by the shell, and it should have been the case here too, but for some reason it isn't. I've worked around it by removing our gradle build script and replacing it so I no longer have this issue.

I just found it surprising because every other shell command "just works", the shell interprets the quotes and removes them according to the normal parsing rules, unless it's an escaped quote.

For example, in bash/zsh `ls -lsa` is the same as `ls -ls'a'` or `ls -l's'a`.

Normally `'` quotes mean you're not allowed to do variable expansion inside these quotes, and `"` mean you are allowed to do variable expansion.

But closing since I no longer have the issue.
